cosign/README.md

```diff
@@ -0,0 +1,163 @@
+    - Timestamps: Time is a critical component of Sigstore. It’s used to verify that a short-lived certificate issued by Fulcio was valid at a previous point, when the artifact was signed.
+
+## Scenarios
+
+`ratify-verifier-go` supports multiple verification scenarios based on different signing methods and trust sources. These scenarios can be categorized into the following main types.
+
+| Scenario             | Purpose                                                                | Use Case                                                                    |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signatures from Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Using an on-disk public key, or a key provided by other providers (e.g., "KMS provider") |
+
+### Cosign Library Keyless Input
+
+1. The Artifact to Verify
+    - The OCI subject.
+
+2. Verification Materials
+    - Identity Parameters: Restrict verification to specific OIDC identities. Valid values include email address, DNS names, IP addresses, and URIs.
+    - OIDC Issuer Parameters: Ensure the certificate was issued by a specific OIDC provider, without Fulcio roots (for BYO PKI).
+
+### Cosign Library Key-based Verify Input
+
+1. The Artifacts to Verify
+    - The OCI subject.
+
+2. Verification Materials
+    - Public Key: The public key in PEM format, corresponding to the private key used to sign the artifact.
+
+NOTE: With OCI artifacts, the signatures and certificates are attached to the OCI artifacts by default.
+
+### Cosign Verify Output
+
+The output format is defined by `ratify-go`, illustrates if the signature is valid and all criteria are met. It includes detailed information such as:
+
+- Error: The error that occurred when the verification failed. If the verification is successful, this field should be nil.
+- Description: Indicates if the verification was successful.
+- Verifier:  The verifier that generated the result.
+- Details(Optional):
+        - artifact_digest: The cryptographic hash of the artifact.
+        - signer_identity: The identity associated with the signing certificate.
+        - signing_cert: Metadata about the Fulcio-issued certificate.
+        - rekor_entry: Confirms whether the signature exists in the transparency log.
+        - timestamp_verified: Ensures that the signature has a valid timestamp.
+
+### Verification Components
```

Comment:
does this section make a difference to the input/ouput/workflow design?

Reply:
This section shows how Cosign separates different functionality of verification based on the current implementation.